modules/tasks/tasks/src/test/java/org/motechproject/tasks/service/impl/TaskTriggerHandlerTest.java

```diff
@@ -784,6 +785,40 @@ public void shouldNotPassFiltersCriteria() throws Exception {
     }
 
     @Test
+    public void shouldPassFiltersCriteriaAndExecuteActions() throws Exception {
+        setTriggerEvent();
+        setActionEvent();
+        setActionFilters();
+
+        when(taskService.findActiveTasksForTriggerSubject(TRIGGER_SUBJECT)).thenReturn(tasks);
+        when(taskService.getActionEventFor(task.getActions().get(0))).thenReturn(actionEvent);
+
+        handler.handle(createEvent());
+
+        verify(taskService).findActiveTasksForTriggerSubject(TRIGGER_SUBJECT);
+        verify(taskService).getActionEventFor(task.getActions().get(0));
+        verify(eventRelay).sendEventMessage(any(MotechEvent.class));
+    }
+
+    @Test
+    public void shouldPassFiltersCriteriaAndNotExecuteActions() throws Exception {
+        setTriggerEvent();
+        setActionEvent();
+        setActionFilters();
+
+        task.getTaskConfig().add(new FilterSet(asList(new Filter("ExternalID (Trigger)", "trigger.externalId", INTEGER, false, EXIST.getValue(), "")),LogicalOperator.AND,1));
+
+        when(taskService.findActiveTasksForTriggerSubject(TRIGGER_SUBJECT)).thenReturn(tasks);
+        when(taskService.getActionEventFor(task.getActions().get(0))).thenReturn(actionEvent);
+
+        handler.handle(createEvent());
+
+        verify(taskService).findActiveTasksForTriggerSubject(TRIGGER_SUBJECT);
+        verify(taskService, never()).getActionEventFor(task.getActions().get(0));
+        verify(eventRelay, never()).sendEventMessage(any(MotechEvent.class));
+    }
```

Comment:
this two methods should test new feature, please add action filter on second action, not the first one